### PR TITLE
Add L2 to `tracr-reverse`

### PR DESCRIPTION
**Pull request by ArthurConmy**

### CODE REVIEW

File: acdc/tracr/utils.py [hunk 1]
```diff
@@ -260,16 +273,26 @@ def get_all_tracr_things(task: Literal["reverse", "proportion"], metric_name: st
         warnings.warn("Test that this only considers the relevant part of the sequence...")
 
         patch_data_tens = data_tens[patch_data_indices]
+
         with torch.no_grad():
-            base_model_logprobs = F.log_softmax(tl_model(data_tens), dim=-1)
+            model_out = tl_model(data_tens)
+            base_model_logprobs = F.log_softmax(model_out, dim=-1) # oh no...
```
Comment (rhaps0dy) [inline acdc/tracr/utils.py:279]: what do you mean "oh no..." ??

**Review by rhaps0dy — APPROVED** [review]
I don't understand the "oh no" comment but code LGTM

**Comment by ArthurConmy** [comment]
Using `log_softmax` is wrong. I wrote in the Exception

> This is wrong - tracr outputs one-hot distributions

i.e the outputs are not logits. So doing softmax is turning [0, 0, 1] into [0.21, 0.21, 0.57] which is not the correct distribution @rhaps0dy

**Comment by rhaps0dy** [comment]
Sure; but if you're gonna take the KL you're treating them as logits anyways. I thought we said we'd do that. Whatever, it's fixed now

**Comment by ArthurConmy** [comment]
I thought that all tracr stuff in the submission would be with L2. 

For the original arXiv paper, I only did one run on both tracr tasks, and just redefined KL divergence (I think this is somewhat cursed now) as something that was only ever 0 or infinity
